flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java

```diff
@@ -228,7 +228,7 @@
 import static org.mockito.Mockito.when;
 
 /** Tests for {@link StreamTask}. */
-public class StreamTaskTest extends TestLogger {
+public class StreamTaskITCase extends TestLogger {
```

Comment:
Which tests are causing trouble? I would guess only one or two? If so, can we maybe extract those offending tests to an `ITCase` file? There is a huge family of various `StreamTask` test files. Having (most of) them in the files named according to the same convention would help avoid future confusions.

Reply:
It should be those that rely on `testRecordWriterClosedOnError`, so `testFailInEndOfConstructor` and the 3 `testRecordWriterClosedOnTransit*` tests.

Reply:
I will look into splitting the test.